components/Booking.php

```diff
@@ -402,7 +402,7 @@ protected function processValidateAfter($validator, $dateTime)
             return $validator->errors()->add('time', lang('igniter.reservation::default.error_invalid_time'));
 
         $autoAllocateTable = (bool)$this->location->getOption('auto_allocate_table', 1);
-        if ($autoAllocateTable AND $this->manager->isFullyBookedOn($dateTime, input('guest')))
+        if ($autoAllocateTable AND ($this->manager->isFullyBookedOn($dateTime, input('guest')) OR !$this->manager->getNextBookableTable($dateTime, input('guest'))))
```

Comment:
`isFullyBookedOn ()` method is same as `! getNextBookableTable ()`.. I'll make the changes